libbeat/docs/shared-faq.asciidoc

```diff
@@ -0,0 +1,43 @@
+//////////////////////////////////////////////////////////////////////////
+//// This content is shared by all Elastic Beats. Make sure you keep the
+//// descriptions here generic enough to work for all Beats that include
+//// this file. When using cross references, make sure that the cross
+//// references resolve correctly for any files that include this one.
+//// Use the appropriate variables defined in the index.asciidoc file to
+//// resolve Beat names: beatname_uc and beatname_lc.
+//// Use the following include to pull this content into a doc file:
+//// include::../../libbeat/docs/shared-faq.asciidoc[]
+//////////////////////////////////////////////////////////////////////////
+
+[[connection-problem]]
+=== Why doesn't my connection to Logstash work?
+
+You may have configured Logstash incorrectly. Make sure that:
+
+* The config file for you Beat specifies the correct port where Logstash is running. See
+{libbeat}/logstash-installation.html#logstash-setup[Setting Up Logstash]. 
+* The Elasticsearch output is commented out in the configuration file and the Logstash output is uncommented. 
+* The most recent Beats input plugin for Logstash is installed. See
+{libbeat}/logstash-installation.html#logstash-input-update[Updating the Beats Input Plugin for Logstash].
+
+[[metadata-missing]]
+=== Why is @metadata missing in Logstash?
+
+Logstash outputs remove `@metadata` fields automatically. Therefore, if Logstash instances are chained directly or via some message
+queue (for example, Redis or Kafka), the `@metadata` field will not be available in the final Logstash instance.
+
+[[diff-logstash-beats]]
+=== What is the difference between Logstash and Beats?
+
+Beats are lightweight data shippers that you install as agents on your servers to send specific types of operational
+data to Elasticsearch. Beats have a small footprint and use fewer system resources than Logstash.
+
+Logstash has a larger footprint, but provides a broad array of input, filter, and output plugins for collecting, enriching,
+and transforming data from a variety of sources. 
+
+For more information, see the https://www.elastic.co/guide/en/logstash/current/introduction.html[Logstash Introduction] and
+the https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[Beats Overview].
+
+// [[ssl-client-fails]]
+// === Why does my SSL client fail to connect
```

Comment:
logstash ssl?

1. check if logstash reachable? `ping`, `telnet`
2. is certificate valid? Do hostname/ip match? One can use `insecure: true` to disable host name checking (not recommended, but okish for testing)
3. Is vertificate valid? Negative number bug in OpenSSL. (Will be 'fixed' when switching to golang 1.6)
4. Is `ssl => true` enabled in logstash beats plugin?